File: java_test/src/DesignPattern/Factory/AbstractFactory/PizzaIngredientFactory.java
package DesignPattern.Factory.AbstractFactory;

/**
 * Created by Administrator on 2018/4/8.
 */

//披萨原料工厂接口
public interface PizzaIngredientFactory {
    public Dough createDough();//创建面团

    public Sauce createSauce();//创建酱油

    public Cheese createCheese();//创建奶酪

    public Veggies[] createVeggies();//创建蔬菜

    public Pepperoni createPepperoni();//创建意大利香肠

    public Clams createClam();//创建蛤蜊
}

File: test_go/src/pattern/factory/abstract_factory/pizza_ingredient_factory.go
package abstract_factory


type PizzaIngredientFactory interface {
	CreateDough() Dough         //创建面团
	CreateSauce() Sauce         //创建酱油
	CreateCheese() Cheese       //创建奶酪
	CreateVeggies() []Veggies   //创建蔬菜
	CreatePepperoni() Pepperoni //创建意大利香肠
	CreateClam() Clams          //创建蛤蜊
}

/*------------------------------------------------------------------------------------------------------------------*/
//1.纽约原料工厂